### PR TITLE
Add utest to the community build

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**
On top #7510.

### CODE REVIEW

**Comment by anatoliykmetyuk** [comment]
```
utest.native[2.11.12,0.3.8].test.testRunnerNative.nativeClang scala.scalanative.build.BuildException: no clang60, clang-6.0, clang50, clang-5.0, clang40, clang-4.0, clang39, clang-3.9, clang38, clang-3.8, clang37, clang-3.7, clang found in $PATH. Install clang (http://www.scala-native.org/en/latest/user/setup.html)
```

@smarter can you please help with getting these installed on the CI machine?

**Comment by smarter** [comment]
I'm on vacation. I suggest disabling the scala-native tests for now, they're unrelated to using utest with dotty anyway.

File: community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala [hunk 1]
```diff
@@ -26,7 +53,7 @@ class CommunityBuildTest {
    *  @param updateCommand  The sbt command used to update the project
    *  @param extraSbtArgs   Extra arguments to pass to sbt
    */
-  def test(project: String, testCommand: String, updateCommand: String, extraSbtArgs: Seq[String] = Nil): Unit = {
+  def test(project: String)(commandAndArgs: => (String, Seq[String])): Unit = {
```
Comment (nicolasstucki) [inline community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala]: Why is commandAndArgs by-name?

File: compiler/src/dotty/tools/dotc/ast/tpd.scala [hunk 2]
```diff
@@ -881,6 +885,10 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
     def appliedToType(targ: Type)(implicit ctx: Context): Tree =
       appliedToTypes(targ :: Nil)
 
+    /** The current tree applied to given type argument: `tree[targ]` */
+    def appliedToTypeTree(targ: Tree)(implicit ctx: Context): Tree =
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/ast/tpd.scala]: Move it between `appliedToTypes` and `appliedToTypeTrees`.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/ast/tpd.scala]: Also look for `appliedToTypeTrees(xyz :: Nil)` in the compiler and replace them with `appliedToTypeTree(xyz)`. I saw some instances in `tpd` and in `QuotesAndSplices`.

File: community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala (hunk 1 above)
Comment (anatoliykmetyuk) [reply community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala]: No particular reason – just following a pattern of implementing control structures (`test`) via curried functions. If it works with by-value, do you think this should be switched to by-value?

Comment (nicolasstucki) [reply community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala]: If there is nothing relying on it I would make it by-value

**Pull request merged**